app/controllers/blogs_controller.rb

```diff
@@ -47,11 +43,17 @@ def destroy
 
   private
 
-  def set_blog
+  def set_current_user_blog
     @blog = current_user.blogs.find(params[:id])
   end
 
+  def set_blog
+    @blog = Blog.viewable(current_user).find(params[:id])
+  end
+
   def blog_params
-    params.require(:blog).permit(:title, :content, :secret, :random_eyecatch)
+    parameters = %i[title content secret]
+    parameters.push(:random_eyecatch) if current_user.premium
```

Comment:
```suggestion
    parameters.push(:random_eyecatch) if current_user.premium?
```

booleanの属性だと?を付けて可読性を上げることが多いですね。

Reply:
確かにtrue/falseが返る場合は、?をつけた方が可読性が上がりそうですね！

?をつけた処理に修正しました。